code/modules/mob/inventory.dm

```diff
@@ -1,178 +1,291 @@
-	if(hand)
-		return put_in_r_hand(W)
-	else
-		return put_in_l_hand(W)
+/mob/proc/put_in_hand(obj/item/I, hand_index)
+	if(!put_in_hand_check(I))
+		return FALSE
+	if(!has_hand_for_held_index(hand_index))
+		return FALSE
+	var/obj/item/curr = held_items[hand_index]
+	if(!curr)
+		I.loc = src
+		held_items[hand_index] = I
+		I.layer = ABOVE_HUD_LAYER
+		I.equipped(src, slot_hands)
+		if(I.pulledby)
+			I.pulledby.stop_pulling()
+		update_inv_hands()
+		I.pixel_x = initial(I.pixel_x)
+		I.pixel_y = initial(I.pixel_y)
+		return hand_index || TRUE
+	return FALSE
+
+
+//Puts the item into the first available left hand if possible and calls all necessary triggers/updates. returns 1 on success.
+/mob/proc/put_in_l_hand(obj/item/I)
+	return put_in_hand(I, get_empty_held_index_for_side("l"))
+
+
+//Puts the item into the first available right hand if possible and calls all necessary triggers/updates. returns 1 on success.
+/mob/proc/put_in_r_hand(obj/item/I)
+	return put_in_hand(I, get_empty_held_index_for_side("r"))
+
+
+/mob/proc/put_in_hand_check(obj/item/I)
+	if(lying && !(I.flags&ABSTRACT))
+		return FALSE
+	if(!istype(I))
+		return FALSE
+	return TRUE
+
+
+//Puts the item into our active hand if possible. returns TRUE on success.
+/mob/proc/put_in_active_hand(obj/item/I)
+	return put_in_hand(I, active_hand_index)
+
 
+//Puts the item into our inactive hand if possible, returns TRUE on success
+/mob/proc/put_in_inactive_hand(obj/item/I)
+	return put_in_hand(I, get_inactive_hand_index())
 
-//Puts the item our active hand if possible. Failing that it tries our inactive hand. Returns 1 on success.
-//If both fail it drops it on the floor and returns 0.
+
+//Puts the item our active hand if possible. Failing that it tries our inactive hand. Returns TRUE on success.
+//If both fail it drops it on the floor and returns FALSE.
 //This is probably the main one you need to know :)
-/mob/proc/put_in_hands(obj/item/W)
-	if(!W)
-		return 0
-	if(put_in_active_hand(W))
-		return 1
-	else if(put_in_inactive_hand(W))
-		return 1
-	else
-		W.loc = get_turf(src)
-		W.layer = initial(W.layer)
-		W.dropped(src)
-		return 0
+/mob/proc/put_in_hands(obj/item/I, del_on_fail = FALSE)
+	if(!I)
+		return FALSE
+	var/hand = get_empty_held_index_for_side("l")
+	if(!hand)
+		hand =  get_empty_held_index_for_side("r")
+	if(hand)
+		if(put_in_hand(I, hand))
+			return TRUE
+	if(del_on_fail)
+		qdel(I)
+		return FALSE
+	I.forceMove(get_turf(src))
+	I.layer = initial(I.layer)
+	I.dropped(src)
+	return FALSE
+
+
+/mob/proc/put_in_hands_or_del(obj/item/I)
+	return put_in_hands(I, TRUE)
 
 
 /mob/proc/drop_item_v()		//this is dumb.
 	if(stat == CONSCIOUS && isturf(loc))
 		return drop_item()
-	return 0
-
-
-//Drops the item in our left hand
-/mob/proc/drop_l_hand()
-	if(!loc || !loc.allow_drop())
-		return
-	return unEquip(l_hand) //All needed checks are in unEquip
+	return FALSE
 
 
-//Drops the item in our right hand
+/*
+/mob/proc/drop_l_hand() //pending re-implementation

```

Comment:
probably not actually, I don't think there's much use for this proc anymore (99% of uses were both procs at once), and I don't think there's use for a proc that drops all left items, or all right items.
tl;dr old comment.